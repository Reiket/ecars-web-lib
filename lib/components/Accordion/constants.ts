export const ACCORDION_TEST_ID = 'accordionTestId';
