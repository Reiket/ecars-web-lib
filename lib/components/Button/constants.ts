import type {ButtonHTMLAttributes} from 'react';

export const BUTTON_SIZE = Object.freeze({
  BIG: 'big',
  SMALL: 'small',
});

export const BUTTON_COLOR = Object.freeze({
  GREEN: 'green',
  WHITE: 'white',
  GRAY: 'gray',
});


export type ButtonAttributesType = ButtonHTMLAttributes<HTMLButtonElement>['type']
export type ButtonSizeType = (typeof BUTTON_SIZE)[keyof typeof BUTTON_SIZE];
export type ButtonColorType = (typeof BUTTON_COLOR)[keyof typeof BUTTON_COLOR];

export const BUTTON_TEST_ID = 'buttonTestId';