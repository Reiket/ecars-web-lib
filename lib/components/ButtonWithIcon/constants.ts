export const LEFT_ICON_BUTTON_TEST_ID = 'leftIconButtonTestId';
export const RIGHT_ICON_BUTTON_TEST_ID = 'rightIconButtonTestId';
