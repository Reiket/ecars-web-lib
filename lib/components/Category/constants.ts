export const CATEGORIES_LIST = Object.freeze({
  GUIDES: 'guides',
  REVIEWS: 'reviews',
  NEWS: 'news',
});

export type CategoriesListType = (typeof CATEGORIES_LIST)[keyof typeof CATEGORIES_LIST];

export const CATEGORY_TEST_ID = 'categoryTestId';
