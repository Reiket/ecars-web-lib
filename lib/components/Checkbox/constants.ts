import type {ElementProps} from '@/services/types';

export interface CheckboxProps extends ElementProps {
  name?: string;
  id?: string;
  disabled?: boolean;
  onChange?: (checked: boolean) => void;
  checked?: boolean;
  hasError?: boolean;
}

export const CHECKBOX_TEST_ID = 'checkboxTestId';
export const CHECKBOX_TEST_INPUT_ID = 'checkboxTestInputId';
