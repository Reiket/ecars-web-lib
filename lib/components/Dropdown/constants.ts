import type {ReactNode} from 'react';

export interface DropdownOption {
  value: string;
  label?: ReactNode;
}

export interface DropdownProps {
  onSelect: (value: string) => void;
  category: string;
  options: DropdownOption[];
}

export const DROPDOWN_TEST_ID = 'dropdownTestId';

export const DROPDOWN_THEME = Object.freeze({
  DARK: 'dark',
  LIGHT: 'light',
});

export type DropdownThemeType = (typeof DROPDOWN_THEME)[keyof typeof DROPDOWN_THEME];
