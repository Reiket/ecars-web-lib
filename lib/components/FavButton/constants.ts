export const FAV_BUTTON_TYPE = Object.freeze({
  TRANSPARENT: 'transparent',
  CIRCLE: 'circle',
});

export type FavButtonType = (typeof FAV_BUTTON_TYPE)[keyof typeof FAV_BUTTON_TYPE];

export const FAV_BUTTON_TEST_ID = 'favButtonTestId';