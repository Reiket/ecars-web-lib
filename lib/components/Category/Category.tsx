import type {FC} from 'react';
import {CategoriesListType, CATEGORY_TEST_ID} from '@/components/Category/constants';
import {cn} from '@/services/helpers';
import {ElementProps} from '@/services/types';

interface Props extends ElementProps {
  category: CategoriesListType;
}

export const Category: FC<Props> = ({category, block}) => {
  const classNames = cn(block, 'category', `category-${category}`);
  return (
    <span
      data-testid={CATEGORY_TEST_ID}
      className={classNames}
    >
      {category}
    </span>
  );
};
