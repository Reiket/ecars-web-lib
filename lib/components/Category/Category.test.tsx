import {cleanup, render, screen} from '@testing-library/react';
import {CATEGORIES_LIST, CATEGORY_TEST_ID} from '@/components/Category/constants';
import {Category} from '@/components/Category/Category';
import '@testing-library/jest-dom';

describe('Category component', () => {
  afterEach(cleanup);
  const categories = Object.values(CATEGORIES_LIST);
  categories.forEach((category) => {
    const testName = `renders category with ${category}-category`;
    test(testName, () => {
      const {container} = render(<Category category={category} />);
      const categoryElement = screen.getByTestId(CATEGORY_TEST_ID);
      expect(categoryElement).toBeInTheDocument();
      expect(categoryElement).toHaveClass(`category-${category}`);
      expect(container).toMatchSnapshot();
    });
  });
});
