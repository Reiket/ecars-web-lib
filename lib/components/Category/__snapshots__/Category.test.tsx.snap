// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Category component > renders category with guides-category 1`] = `
<div>
  <span
    class="category category--guides"
    data-testid="categoryTestId"
  >
    Guides
  </span>
</div>
`;

exports[`Category component > renders category with news-category 1`] = `
<div>
  <span
    class="category category--news"
    data-testid="categoryTestId"
  >
    News
  </span>
</div>
`;

exports[`Category component > renders category with reviews-category 1`] = `
<div>
  <span
    class="category category--reviews"
    data-testid="categoryTestId"
  >
    Reviews
  </span>
</div>
`;

exports[`Category component renders category with guides-category 1`] = `
<div>
  <span
    class="category category--guides"
    data-testid="categoryTestId"
  >
    Guides
  </span>
</div>
`;

exports[`Category component renders category with news-category 1`] = `
<div>
  <span
    class="category category--news"
    data-testid="categoryTestId"
  >
    News
  </span>
</div>
`;

exports[`Category component renders category with reviews-category 1`] = `
<div>
  <span
    class="category category--reviews"
    data-testid="categoryTestId"
  >
    Reviews
  </span>
</div>
`;
